### PR TITLE
Add optional labels to date range picker

### PR DESCRIPTION
**Pull request by hututudor**
https://etvas.atlassian.net/browse/ETV-4410

### CODE REVIEW

File: src/components/DatePicker/DateRangePicker.jsx [hunk 1]
```diff
@@ -229,7 +250,10 @@ DateRangePicker.propTypes = {
   displayFormat: PropTypes.string,
   label: PropTypes.node,
   placeholder: PropTypes.node,
-  navigationByYear: PropTypes.bool
+  navigationByYear: PropTypes.bool,
+  labelYear: PropTypes.string,
+  labelStartDate: PropTypes.string,
+  labelEndDate: PropTypes.string
```
Comment (sergiuelisei) [inline src/components/DatePicker/DateRangePicker.jsx:256]: Mayne we will not use this only to pass strings, I think `PropTypes.node` is a better option

File: src/components/DatePicker/Years.jsx [hunk 2]
```diff
@@ -92,7 +96,8 @@ Years.propTypes = {
   displayStart: PropTypes.number,
   displayEnd: PropTypes.number,
   displayStatus: PropTypes.func,
-  onChange: PropTypes.func
+  onChange: PropTypes.func,
+  label: PropTypes.string
```
Comment (sergiuelisei) [inline src/components/DatePicker/Years.jsx:100]: Maybe we will not use this only to pass strings, I think PropTypes.node is a better option

Comment (hututudor) [reply src/components/DatePicker/Years.jsx]: 6693dd4c5757ae26b356ee7422623c258ee968e5

File: src/components/DatePicker/DateRangePicker.jsx (hunk 1 above)
Comment (hututudor) [reply src/components/DatePicker/DateRangePicker.jsx]: 6693dd4c5757ae26b356ee7422623c258ee968e5

**Review by sergiuelisei — APPROVED** [review]

File: src/components/DatePicker/Calendar.jsx [hunk 3]
```diff
@@ -217,6 +221,9 @@ export const Calendar = ({
 
   return (
     <Flex justifyContent='flex-start' flexWrap='wrap' width='224px' {...props}>
+      <Typography variant='labelSmallBold' mb={1}>
+        {label}
+      </Typography>
```
Comment (alsofronie) [inline src/components/DatePicker/Calendar.jsx:226]:
```suggestion
      {label ? (<Typography variant='labelSmallBold' mb={1}>
        {label}
      </Typography>): null }
```

File: src/components/DatePicker/Years.jsx [hunk 4]
```diff
@@ -44,6 +45,9 @@ export const Years = ({
 
   return (
     <Box width='224px' {...props}>
+      <Typography variant='labelSmallBold' px={3} mb={1}>
+        {label}
+      </Typography>
```
Comment (alsofronie) [inline src/components/DatePicker/Years.jsx:50]:
```suggestion
      {label ? (<Typography variant='labelSmallBold' px={3} mb={1}>
        {label}
      </Typography>): null}
```

**Review by alsofronie — CHANGES_REQUESTED** [review]
We need to test for label in order to avoid an empty div with margins (which will take space)

**Review by alsofronie — APPROVED** [review]
